src/packs.c

```diff
@@ -52,19 +52,31 @@ static int download_pack(int oldversion, int newversion, char *module)
 		return 0;
 	}
 
-	string_or_die(&url, "%s/%i/pack-%s-from-%i.tar", content_url, newversion, module, oldversion);
-
-	err = swupd_curl_get_file(url, filename, NULL, NULL, true);
-	if (err) {
+	if (is_mix) {
+		string_or_die(&url, "%s/%i/pack-%s-from-%i.tar", MIX_STATE_DIR, newversion, module, oldversion);
+		err = link(url, filename);
+		printf("Linked %s to %s\n", url, filename);
+		if (err) {
+			free(filename);
+			free(url);
+			return err;
+		}
 		free(url);
-		if ((lstat(filename, &stat) == 0) && (stat.st_size == 0)) {
-			unlink(filename);
+	} else {
+		string_or_die(&url, "%s/%i/pack-%s-from-%i.tar", content_url, newversion, module, oldversion);
+
+		err = swupd_curl_get_file(url, filename, NULL, NULL, true);
+		if (err) {
+			free(url);
+			if ((lstat(filename, &stat) == 0) && (stat.st_size == 0)) {
```

Comment:
My guess is so the next time it will try getting the file again. see comment on line 91